File: 383-ransom-note/383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] charSet = new int[26]; 
        for(char c : magazine.toCharArray()){
            charSet[c-'a']++;
        }
        for(char c : ransomNote.toCharArray()){
            if(--charSet[c-'a'] < 0) return false;
        }
        return true;
    }
}

File: 383-ransom-note/383-ransom-note.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    const map = {};
    for(let char of magazine){
        if(!map[char]){
            map[char] = 0;
        }
        map[char]++;
    }
    for(let char of ransomNote){
        if(!map[char]){
            return false;
        }
        map[char]--;
    }
    return true;
};